molgenis-api/src/main/resources/SelectionParser.jj

```diff
@@ -22,11 +22,12 @@ TOKEN: {
 	| <#LETTER: ["a"-"z","A"-"Z"]>
   |	<#DIGIT: ["0"-"9"]>
   |	<UNDERSCORE: "_">
+  | <DASH: "-">
   |	<HASH: "#">
 }
 
 TOKEN: {
-		<IDENTIFIER: (<LETTER> | <DIGIT> | <UNDERSCORE> | <HASH>)+>
+		<IDENTIFIER: (<LETTER> | <DIGIT> | <UNDERSCORE> | <DASH> | <HASH>)+>
```

Comment:
i thought the dash was only allowed in-between other tokens

Reply:
@connoratrug Technically, a dash is only allowed at the end of an attribute name if it is followed by two letters. There are more attribute naming rules that are not being checked by the above expression: names can't start with an underscore or hash for example... 

Invalid names won't break anything, the API knows how to deal with them. That's why I decided to keep this change small.